components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/tests/Controller/DmApiPublisherMethodControllerTests.cs

```diff
@@ -0,0 +1,200 @@
+            legacyCliModelProviderMock.Setup(p => p.LegacyCliModel).Returns(legacyCliModel);
+            agentConfigProviderMock.Setup(p => p.Config).Returns(new AgentConfigModel());
+
+            var orchestrator = new LegacyJobOrchestrator(publishedNodesJobConverter, legacyCliModelProviderMock.Object, agentConfigProviderMock.Object, jobSerializer, TraceLogger.Create(), identityMock.Object);
+            var methodsController = new PublisherMethodsController(orchestrator);
+
+            using var publishPayloads = new StreamReader(publishedNodesFile);
+            var publishNodesRequest = newtonSoftJsonSerializer.Deserialize<List<PublishNodesRequestApiModel>>(await publishPayloads.ReadToEndAsync().ConfigureAwait(false));
+
+            foreach (var request in publishNodesRequest) {
+                var initialNode = request.OpcNodes.First();
+                for (int i = 0; i < 10000; i++) {
+                    request.OpcNodes.Add(new PublishedNodeApiModel {
+                        Id = initialNode.Id + i.ToString(),
+                        DataSetFieldId = initialNode.DataSetFieldId,
+                        DisplayName = initialNode.DisplayName,
+                        ExpandedNodeId = initialNode.ExpandedNodeId,
+                        HeartbeatIntervalTimespan = initialNode.HeartbeatIntervalTimespan,
+                        OpcPublishingInterval = initialNode.OpcPublishingInterval,
+                        OpcSamplingInterval = initialNode.OpcSamplingInterval,
+                        QueueSize = initialNode.QueueSize,
+                        SkipFirst = initialNode.SkipFirst,
+                    });
+                }
+
+                var publishNodesResult = await FluentActions
+                    .Invoking(async () => await methodsController.PublishNodesAsync(request).ConfigureAwait(false))
+                    .Should()
+                    .NotThrowAsync()
+                    .ConfigureAwait(false);
+
+                publishNodesResult.Subject.StatusMessage.First()
+                    .Should()
+                    .Be("Succeeded");
+
+            }
+
+            var tasks = new List<Task<JobProcessingInstructionModel>>();
+            for (var i = 0; i < 10; i++) {
+                tasks.Add(orchestrator.GetAvailableJobAsync(i.ToString(), new JobRequestModel()));
+            }
+
+            tasks.Where(t => t.Result != null)
+                .Select(t => t.Result.Job.JobConfiguration)
+                .Distinct().Count()
+                .Should()
+                .Be(2);
+
+            using var unpublishPayloads = new StreamReader(publishedNodesFile);
+            var unpublishNodesRequest = newtonSoftJsonSerializer.Deserialize<List<PublishNodesRequestApiModel>>(await unpublishPayloads.ReadToEndAsync().ConfigureAwait(false));
+            foreach (var request in unpublishNodesRequest) {
+                var initialNode = request.OpcNodes.First();
+                for (int i = 0; i < 10000; i++) {
```

Comment:
The same request has been constructed for publishing nodes, we can reuse it.

Reply:
right, that was a forgotten leftover from an attempt of multiplying the request entries as well.